src/main/bash/NodeLimit/limit.sh

```diff
@@ -3,33 +3,40 @@
-    time_values=$(grep -Po "$no_quotes_regex" <<< "$time_values") #Get rid of quotes
-    IFS=_ read -r quota_us period_us <<< "$time_values" #Split both time values based on "_" separator
-
-    echo "Read tag: $node_tag"
-    echo "Read Quota: $quota_us"
-    echo "Read Period: $period_us"
-
-    #To limit the number of cores assigned cfs_period and cfs_quota have to be set
-    # number of cores = cfs_quota_us / cfs_period_us (e.g. 0.5 core = 50000 / 100000, 2 cores = 200000 / 100000)
-    sudo cgcreate -g cpu:/"$node_tag" #Create a cpu limited group with the name of the node
-    sudo cgset -r cpu.cfs_period_us="$period_us" "$node_tag"
-    sudo cgset -r cpu.cfs_quota_us="$quota_us" "$node_tag"
-
-    #Assignment of the processes to the group based on the PID
-    #First command in the pipe (ps) outputs all running processes (-e), in full format(-f) and the threads (-T) with SPID
-    #AWK command, returns the SPIDs of the process which were called with the node_tag as argument
-    # Xargs turns the output of awk into arguments for cgclassify, which then assigns the processes to the limited group
-    ps -ef -T | awk -v tag="$node_tag" '!/awk/ && $0 ~ tag { print $3 }' | xargs -t sudo cgclassify -g cpu:/"$node_tag"
-
-  done < limitFile
+      curl -s "$1" | grep -Po '"\w+":"\w+"' > limitFile #Do the GET request and save the data to a file
+      if [[ "$read_data" != $(cat limitFile) ]] #Only process if there is new data
+      then
+        while IFS=: read -r node_tag time_values #For each line of the file
+        do
+          node_tag=$(grep -Po "$no_quotes_regex" <<< "$node_tag") #Get rid of quotes
+          time_values=$(grep -Po "$no_quotes_regex" <<< "$time_values") #Get rid of quotes
+          IFS=_ read -r quota_us period_us <<< "$time_values" #Split both time values based on "_" separator
+
+          echo "Read tag: $node_tag"
+          echo "Read Quota: $quota_us"
+          echo "Read Period: $period_us"
+
+          #To limit the number of cores assigned cfs_period and cfs_quota have to be set
+          # number of cores = cfs_quota_us / cfs_period_us (e.g. 0.5 core = 50000 / 100000, 2 cores = 200000 / 100000)
+          sudo cgcreate -g cpu:/"$node_tag" #Create a cpu limited group with the name of the node
+          sudo cgset -r cpu.cfs_period_us="$period_us" "$node_tag"
+          sudo cgset -r cpu.cfs_quota_us="$quota_us" "$node_tag"
+
+          #Assignment of the processes to the group based on the PID
+          #First command in the pipe (ps) outputs all running processes (-e), in full format(-f) and the threads (-T) with SPID
+          #AWK command, returns the SPIDs of the process which were called with the node_tag as argument
+          # Xargs turns the output of awk into arguments for cgclassify, which then assigns the processes to the limited group
+          ps -ef -T | awk -v tag="$node_tag" '!/awk/ && $0 ~ tag { print $3 }' | xargs -t sudo cgclassify -g cpu:/"$node_tag"
```

Comment:
A small problem here, if the process to be limited is not yet created (or the application restarted) the limit will not be applied.

This line must be executed even if there were no changes in the limits, it must be excluded from the if condition.

Reply:
I thought about that but I wasnt sure because, is it right to be constantly adding (each loop of the infinite while) the same (most of the time) processes to the group ?

Reply:
You should always add a delay to the loop, lets say 1 second.